lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -178,8 +183,8 @@ def refresh
     end
 
     def kill
-      if @session
-        run_sync("pkill -f #{remote_command_file('script')}")
+      if @pid_path
```

Comment:
`@pid_path` will almost always be set, it is by no means a good representation of whether something runs on the remote end or not. In this case I'd vote for a default implementation using `@session` for the regular script runner and then override it with unconditional remote kill for the polling runner.